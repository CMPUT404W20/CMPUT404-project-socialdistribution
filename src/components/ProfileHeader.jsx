import React, { Component } from "react";
import "../styles/ProfileHeader.scss";

class ProfileHeader extends Component {
  constructor(props) {
    super(props);
    this.state = {
    };
  }

  render() {
    return (
      <div className="profileHeader" />
    );
  }
}
export default ProfileHeader;
