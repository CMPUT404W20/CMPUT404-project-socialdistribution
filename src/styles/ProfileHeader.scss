$header-width: 100%;
$header-height: 250px;
$bg-path: "../images/profile-bg.png";
$text-color: #181818;
$button-border-color: #dadada;
.profileHeader {
    margin-top: 90px;
    background-color: white;
    margin-bottom: 50px;
    width: $header-width;
    height: $header-height;
}

.image-section {
    height: 50%;
    background: url($bg-path) bottom;
    background-size: auto 100%;
    .edit-profile-button {
        float: right;
        display: inline-flex;
        justify-content: center;
        align-items: center;
        background: rgba(255, 255, 255, 0.9);
        border-radius: 4px;
        border: 1px solid $button-border-color;
        -webkit-appearance: none;
        -moz-appearance: none;
        height: 1.8rem;
        margin-top: 0.9rem;
        margin-right: 0.9rem;
        padding-left: 0.9rem;
        padding-right: 0.9rem;
        font-style: normal;
        color: $text-color;
        font-size: 7pt;
        text-align: center;
        text-transform: uppercase;
    }
    .edit-icon {
        margin-right: 2px;
        color: $text-color;
        height: 15pt;
    }
}

.row1 {
    margin-top: 1.5rem;
    display: inline-flex;
    padding-left: 3rem;
    padding-right: 3rem;
    width: 100%;
    align-items: center;
    justify-content: space-between;
    font-size: 9pt;
     :first-child {
        font-style: normal;
        font-weight: bold;
        text-transform: uppercase;
        font-size: 11pt;
    }
}