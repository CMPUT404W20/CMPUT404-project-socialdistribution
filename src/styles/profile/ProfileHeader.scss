@import "../common.scss";
$header-width: 100%;
$header-height: 250px;
$bg-path: "../../images/profile-bg.png";
.profileHeader {
    background-color: white;
    padding-bottom: 1rem;
    width: $header-width;
    height: $header-height;
    border-radius: $radius;
    overflow: hidden;
}

.image-section {
    height: 50%;
    background: url($bg-path) bottom;
    background-size: auto 100%;
}

.row1 {
    margin-top: 1.5rem;
    display: inline-flex;
    padding-left: 3rem;
    padding-right: 3rem;
    width: 100%;
    align-items: center;
    justify-content: space-between;
    font-size: 9pt;
     :first-child {
        font-style: normal;
        font-weight: bold;
        text-transform: uppercase;
        font-size: 11pt;
    }
}

.row2 {
    float: right;
    padding-right: 3rem;
    button {
        @extend %button;
        width: 10rem;
        display: inline-flex;
        justify-content: center;
        align-items: center;
    }
    .edit-icon {
        margin-right: 2px;
        color: white;
        height: 15pt;
    }
    #friend-status {
        @extend %button;
        &:not(:disabled) {
            &:hover,
            &:focus {
                box-shadow: none;
            }
        }
        cursor: default;
    }
    .dropdown-menu {
        @extend %dropdown;
    }
}