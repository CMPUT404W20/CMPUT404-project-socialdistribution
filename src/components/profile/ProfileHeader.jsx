import React, { Component } from "react";
import "../../styles/profile/ProfileHeader.scss";
import EditOutlinedIcon from "@material-ui/icons/EditOutlined";
import Dropdown from "react-bootstrap/Dropdown";
import DropdownButton from "react-bootstrap/DropdownButton";
import EditProfileModal from "./EditProfileModal";

class ProfileHeader extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: "Username",
      remote: false,
      friend: true,
      following: true,
      isSelf: true,
      // isSelf will be changed by comparing the userID and the loggedIn user's id
      // to do: fetch the user
      modalShow: false,
    };
  }

  handleEditButtonClick = () => {
    // eslint-disable-next-line no-alert
    alert("todo");
  };

  renderModal = () => {
    const { modalShow } = this.state;
    if (modalShow) {
      this.setState({ modalShow: false });
    } else {
      this.setState({ modalShow: true });
    }
  };

  renderStatus = () => {
    const { isSelf, following, friend } = this.state;
    if (!isSelf) {
      if (!friend && !following) {
        return (this.renderFollowButton());
      }
      if (!friend && following) {
        return (this.renderDropDown(false));
      }
      return (this.renderDropDown(true));
    }
    return (this.renderEditProfileButton());
  };

  renderEditProfileButton = () => (
    <button
      type="button"
      className="edit-profile-button"
      onClick={this.renderModal}
    >
      <EditOutlinedIcon className="edit-icon" />
      <span>EDIT PROFILE</span>
    </button>
  );

  handleFollow = () => {
    this.setState({ following: true });
  };

  renderDropDown = (isFriend) => (
    <DropdownButton
      id="friend-status"
      title={isFriend === true ? "FRIENDS" : "FOLLOWING"}
      drop="down"
      alignRight
    >
      <Dropdown.Item onClick={this.handleUnFollow}>{isFriend === true ? "Unfriend" : "Unfollow"}</Dropdown.Item>
    </DropdownButton>
  );

  handleUnFollow = () => {
    this.setState({ friend: false, following: false });
  };

  renderFollowButton = () => (
    <button
      type="button"
      className="follow-button"
      onClick={this.handleFollow}
    >
      Follow
    </button>
  );

  render() {
    const {
      username, remote, modalShow, isSelf,
    } = this.state;
    return (
      <div className="profileHeader">
        <div className="image-section" />
        <div className="user-section">
          <div className="row1">
            <p>{username}</p>
            {isSelf ? null : (<p>{remote === true ? "Remote" : "Local"}</p>)}
          </div>
          <div className="row2">
            {this.renderStatus()}
            <EditProfileModal show={modalShow} onHide={this.renderModal} />
          </div>
        </div>
      </div>
    );
  }
}
export default ProfileHeader;
