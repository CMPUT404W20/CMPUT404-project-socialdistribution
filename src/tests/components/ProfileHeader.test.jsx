import React from "react";
import { shallow } from "enzyme";
import ProfileHeader from "../../components/profile/ProfileHeader";

describe("Navigation bar Component", () => {
  it("should render correctly", () => {
    const component = shallow(<ProfileHeader isFriends isFollowing isSelf host="" username="aaa" />);
    expect(component).toMatchSnapshot();
  });
});
