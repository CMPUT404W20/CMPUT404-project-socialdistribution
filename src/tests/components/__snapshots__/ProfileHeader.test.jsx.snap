// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation bar Component should render correctly 1`] = `
<div
  className="profileHeader"
>
  <div
    className="image-section"
  />
  <div
    className="user-section"
  >
    <div
      className="row1"
    >
      <p>
        Username
      </p>
      <p>
        Local
      </p>
    </div>
    <div
      className="row2"
    >
      <button
        className="edit-profile-button"
        onClick={[Function]}
        type="button"
      >
        <EditOutlinedIcon
          className="edit-icon"
        />
        <span>
          EDIT PROFILE
        </span>
      </button>
      <EditProfileModal
        onHide={[Function]}
        show={false}
      />
    </div>
  </div>
</div>
`;
